src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdkAppEngineFlexibleStaging.java

```diff
@@ -114,13 +110,14 @@ static void copyDockerContext(StageFlexibleConfiguration config, CopyService cop
     }
   }
 
+  @VisibleForTesting
   static void copyAppEngineContext(StageFlexibleConfiguration config, CopyService copyService)
       throws IOException {
     // Copy app.yaml and other App Engine config files to staging
     String[] appEngineConfigFiles = config.getAppEngineDirectory().list();
     if (appEngineConfigFiles != null) {
       for (String configFile : appEngineConfigFiles) {
-        if (APP_ENGINE_CONFIG_FILES_WHITELIST.contains(configFile)) {
+        if (APP_YAML.equals(configFile)) {
           copyService.copyFileAndReplace(
               config.getAppEngineDirectory().toPath().resolve(configFile),
               config.getStagingDirectory().toPath().resolve(configFile));
```

Comment:
line 128 needs to be modified, we're don't fail if we find "unexpected" configuration files. We just copy app.yaml from that directory.

I think this whole loop might need to change, no more loop?